packages/core/http/core-http-server-internal/src/http_config.ts

```diff
@@ -205,10 +205,10 @@ const configSchema = schema.object(
         },
       }
     ),
-    // allow access to internal routes by default to prevent breaking changes in current offerings
+    // disable access to internal routes by default
     restrictInternalApis: offeringBasedSchema({
-      serverless: schema.boolean({ defaultValue: false }),
-      traditional: schema.boolean({ defaultValue: false }),
+      serverless: schema.boolean({ defaultValue: true }),
```

Comment:
Changing the default also meant updating tests to either disable the restriction or add the header required for internal requests.